langwatch/src/components/prompt-configs/hooks/usePromptConfigForm.ts

```diff
@@ -0,0 +1,138 @@
+import { z } from "zod";
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useOrganizationTeamProject } from "~/hooks/useOrganizationTeamProject";
+import { toaster } from "~/components/ui/toaster";
+import { usePromptConfigVersionMutation } from "./usePromptConfigVersionMutation";
+import { api } from "~/utils/api";
+import { useEffect } from "react";
+import type { LlmPromptConfig, LlmPromptConfigVersion } from "@prisma/client";
+
+const promptConfigSchema = z.object({
+  name: z.string().min(1, "Name is required"),
+});
+
+const versionSchema = z.object({
```

Comment:
This is what can change. Right now we are only supporting the 1 version, but we should create a library of this to handle correctly.